utils/posts.ts

```diff
@@ -0,0 +1,37 @@
+import { extract } from "std/encoding/front_matter/yaml.ts";
+import { join } from "std/path/mod.ts";
+
+export interface Post {
+  slug: string;
+  title: string;
+  publishedAt: Date;
+  content: string;
+  abstract: string;
```

Comment:
```suggestion
  summary: string;
```
Some people, especially non-native English speakers, may find abstract confusing.